File: Array/21_30/23Solution.py
def maximumUnits(boxTypes,truckSize):
        boxTypes=sorted(boxTypes,key=lambda x:-x[1])
        ans=0
        for i in boxTypes:
            if truckSize >= i[0]:
                ans += i[0] * i[1]
                truckSize -= i[0]
                continue
            else:
                ans += truckSize * i[1]
                break
        return ans
n=int(input())
truckSize=int(input())
boxTypes=[]
for i in range(0, n):
    ele = [int(input()), int(input())]
    boxTypes.append(ele)
print(maximumUnits(boxTypes,truckSize))

File: Array/21_30/23Solution.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes,(a,b)->Integer.compare(b[1],a[1]));
        
        int units=0;
        for(int[] box:boxTypes){
            if(truckSize>box[0]){
                units+=(box[0]*box[1]);
                truckSize-=box[0];
            }else{
                units+=truckSize*box[1];
                return units;
            }
        }
        
        return units;
    }
}
